src/components/Calculator.js

```diff
@@ -1,41 +1,109 @@
-const Calculator = () => (
-  <div className="calculator-container">
-    <h2 className="calculator-title">Calculator App</h2>
-    <ChildComponent />
-  </div>
-);
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
+
+const Calculator = () => {
+  const [data, setData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (buttonName) => {
+    setData(calculate(data, buttonName));
+  };
 
-function ChildComponent() {
+  return (
+    <div className="calculator-container">
+      <h2 className="calculator-title">Calculator App</h2>
+      <ChildComponent data={data} handleClick={handleClick} />
+    </div>
+  );
+};
+
+function ChildComponent({ data, handleClick }) {
   return (
     <div className="main-container">
-      <input className="input-screen height" type="text" id="input" name="name" pattern="[0-9+\-*/()\s]+" placeholder="0" />
-      <div className="buttons-container">
-        <button className="button math-operation height" type="button">AC</button>
-        <button className="button math-operation height" type="button">+/-</button>
-        <button className="button math-operation height" type="button">%</button>
-        <button className="button math-operation orange height" type="button">+</button>
-        <br />
-        <button className="button math-number height" type="button">7</button>
-        <button className="button math-number height" type="button">8</button>
-        <button className="button math-number height" type="button">9</button>
-        <button className="button math-operation orange height" type="button">x</button>
-        <br />
-        <button className="button math-number height" type="button">4</button>
-        <button className="button math-number height" type="button">5</button>
-        <button className="button math-number height" type="button">6</button>
-        <button className="button math-operation orange height" type="button">-</button>
-        <br />
-        <button className="button math-number height" type="button">1</button>
-        <button className="button math-number height" type="button">2</button>
-        <button className="button math-number height" type="button">3</button>
-        <button className="button math-operation orange height" type="button">+</button>
-        <br />
-        <button className="button-zero math-number height" type="button">0</button>
-        <button className="button math-number height" type="button">.</button>
-        <button className="button math-operation orange height" type="button">=</button>
+      <input
+        className="input-screen height"
+        type="text"
+        id="input"
+        name="name"
+        pattern="[0-9+\-*/()\s]+"
+        placeholder="0"
+        value={typeof data.next === 'number' ? data.next : parseFloat(data.next) || data.total || '0'}
```

Comment:
[Otional] In order to know what operation is being performed, I think you could make the "operation" state as part of the display. For example:

```javascript
  { total, operation, next } = data
  value={`${total === null ? 0 : total} ${operation || ''} ${next || ''}`}
```